src/usecases/verifyTrustAdminToken.test.js

```diff
@@ -65,4 +67,55 @@ describe("verifyTrustAdminToken", () => {
       Location: "/trust-admin/login",
     });
   });
+
+  it("updates the cookie if the token is regenerated", () => {
+    const regenRes = {
+      writeHead: jest.fn(() => regenRes),
+      end: jest.fn(),
+      setHeader: jest.fn(),
+    };
+    const regenReq = {
+      headers: {
+        cookie: "token=sample.token.value",
+      },
+    };
+
+    const callback = jest.fn();
+    const authenticationToken = {
+      type: "trustAdmin",
+    };
+    const tokenProvider = {
+      validate: jest.fn(() => authenticationToken),
+    };
+
+    const regeneratedToken = { type: "trustAdmin", test: "1" };
+
+    const container = {
+      getTokenProvider: () => tokenProvider,
+      getRegenerateToken: () =>
+        jest.fn().mockReturnValue({
+          isTokenRegenerated: true,
+          regeneratedEncodedToken: "encodedToken",
+          regeneratedToken: regeneratedToken,
+        }),
+    };
+
+    verifyTrustAdminToken(callback)({
+      req: regenReq,
+      res: regenRes,
+      container,
+    });
+
+    expect(regenRes.setHeader).toHaveBeenCalledWith("Set-Cookie", [
+      "token=encodedToken; httpOnly; path=/;",
+    ]);
+
+    expect(regenRes.setHeader).toHaveBeenCalled();
+
+    expect(callback).toHaveBeenCalledWith(
+      expect.objectContaining({
+        authenticationToken: regeneratedToken,
+      })
+    );
```

Comment:
Do you think it's worth adding an expectation to check that the `regenerateToken` usecase has been called with the original `authenticationToken`?